packages/react-core/src/components/Toolbar/Toolbar.tsx

```diff
@@ -129,5 +129,9 @@ export class Toolbar extends React.Component<ToolbarProps, ToolbarState> {
         </ToolbarContext.Provider>
       </div>
     );
+  };
+
+  render() {
+    return <GenerateId>{randomId => this.renderToolbar(this.props.id || randomId)}</GenerateId>;
```

Comment:
I find this a little confusing.  GenerateId/randomId can be an id provided by app dev in props and not random or generated?

Reply:
It's fair it's a little confusing...
Basically renderToolbar needs an ID, by wrapping the call in a GenerateId component, I can retrieve a randomID, or I can use the one passed by the consumer. would this be better?

```return this.props.id ? this.renderToolbar(this.props.id) : <GenerateId>{randomId => this.renderToolbar(randomId)</GenerateId>;```